docs/spark/tutorials/batch-processing.md

```diff
@@ -0,0 +1,150 @@
+
+## Download the sample data
+
+[GHTorrent](http://ghtorrent.org/) monitors all public GitHub events, such as info about projects, commits, and watchers, and stores the events and their structure in databases. Data collected over different time periods is available as downloadable archives. Because the dump files are very large, this guide uses a [truncated version of the dump file](https://github.com/dotnet/spark/tree/master/examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/projects_smaller.csv) that can be downloaded from GitHub.
+
+## Create a console application
+
+1. In your command prompt, run the following commands to create a new console application:
+
+   ```console
+   dotnet new console -o mySparkBatchApp
+   cd mySparkBatchApp
+   ```
+
+   The `dotnet` command creates a `new` application of type `console` for you. The `-o` parameter creates a directory named *mySparkBatchApp* where your app is stored and populates it with the required files. The `cd mySparkBatchApp` command changes the directory to the app directory you just created.
+
+1. To use .NET for Apache Spark in an app, install the Microsoft.Spark package. In your console, run the following command:
+
+   ```console
+   dotnet add package Microsoft.Spark
+   ```
+
+## Create a SparkSession
+
+1. Add the following additional `using` statements to the top of the *Program.cs* file in *mySparkBatchApp*:
+
+   ```csharp
+   using System;
+   using Microsoft.Spark.Sql;
+   using static Microsoft.Spark.Sql.Functions;
+   ```
+
+1. Add the following code to establish a new SparkSession. The SparkSession is the entry point to programming Spark with the Dataset and DataFrame API. By calling the `spark` object, you can access Spark and DataFrame functionality throughout your program.
+
+   ```csharp
+   SparkSession spark = SparkSession
+        .Builder()
+        .AppName("GitHub and Spark Batch")
+        .GetOrCreate();
+   ```
+
+## Prepare the data
+
+1. Read the input file into a DataFrame, which is a distributed collection of data organized into named columns. You can set the columns for your data through <xref:Microsoft.Spark.Sql.DataFrame.Schema%2A>. Use <xref:Microsoft.Spark.Sql.DataFrame.Show%2A> to display the data in your DataFrame. Be sure to update the CSV filepath to the location of the GitHub data you downloaded.
+
+   ```csharp
+   DataFrame projectsDf = spark
+       .Read()
+       .Schema("id INT, url STRING, owner_id INT, " +
+       "name STRING, descriptor STRING, language STRING, " +
+       "created_at STRING, forked_from INT, deleted STRING" +
+       "updated_at STRING")
+       .Csv("filepath");
+
+   projectsDf.Show();
+   ```
+
+1. Use <xref:Microsoft.Spark.Sql.DataFrame.Na%2A> to drop rows with NA (null) values, and the <xref:Microsoft.Spark.Sql.DataFrame.Drop%2A> method to remove certain columns from your data. This helps prevent errors if you try to analyze null data or columns that are not relevant to your final analysis.
```

Comment:
Although the two uses of `xref` appear strange in the markdown, I assume they are normal-linking links/references when the doc gets published?